routes/search.js

```diff
@@ -0,0 +1,41 @@
+var express = require('express');
+var Reviews = require ('../database/reviews');
+var router = express.Router();
+
+router.route('/')
+
+	.get(function (req, res) {
+		if(req.query.name || req.query.placeType || req.query.stars){
+			if(req.query.stars == ''){
+				req.query.stars = {$gt: '0'};
+			}
+			Reviews.find({'name' : {$regex: req.query.name, $options: 'i'}, 'placeType' : {$regex: req.query.placeType, $options: 'i'}, 'stars' : req.query.stars}, function (err, reviews) {
```

Comment:
Ici on a un souci.

1. T'aurais pu construire un objet query afin d'éviter passer toujours les 3 attributs dans la query
	var query = {};
	if (req.query.name) {
		query.name = {$regex: req.query.name, $options: 'i'};
	}
	if (req.query.placeType) {
		query.placeType = {$regex: req.query.placeType, $options: 'i'};
	}
	if (req.query.stars) {
		query.stars = req.query.stars;
	}
	
Le souci avec cette requête est que tu obliges à avoir les 3 paramètres pour chercher un lieu mais si on veut juste filtrer par nom, on ne peut pas le faire. 

Si le body vient d'une requête faite à partir de la view html, on n'a pas ce souci.
Mais si la requete vient d'un client REST et je tape localhost:3000/search?name=McDo je vais avoir une exception car je passe un null dans le requete  (req.query.placeType is undefined)
			
Donc en construisant l'objet pour la query comme le code que je t'ai mis en haut, tu peux remplacer par
Reviews.find(query, function (err, reviews) {